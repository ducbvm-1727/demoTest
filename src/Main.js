import React, { Component } from 'react'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

export default class Main extends Component {
  render() {
    return (
      <View>
        <Text style={styles.text}>{'My name is ' + `${this.props.name}`}</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  text: {
    fontSize: 28,
    alignItems: 'center',
    justifyContent: 'center'
  }
})
