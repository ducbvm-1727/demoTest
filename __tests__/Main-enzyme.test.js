import React from 'react'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { shallow } from 'enzyme'
import Main from '../src/Main'
Enzyme.configure({ adapter: new Adapter() });

it('Main test Enzyme', () => {
  const snap = shallow(
    <Main name={'Duc'} />
  )
  expect(snap).toMatchSnapshot()

  snap.setProps({ name: 'Duc Dep trai' });

  expect(snap).toMatchSnapshot();
})