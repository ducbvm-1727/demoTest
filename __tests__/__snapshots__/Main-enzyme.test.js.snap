// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main test Enzyme 1`] = `
<View>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 28,
        "justifyContent": "center",
      }
    }
  >
    My name is Duc
  </Text>
</View>
`;

exports[`Main test Enzyme 2`] = `
<View>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 28,
        "justifyContent": "center",
      }
    }
  >
    My name is Duc Dep trai
  </Text>
</View>
`;
